net/haproxy/src/opnsense/mvc/app/controllers/OPNsense/HAProxy/forms/dialogAction.xml

```diff
@@ -23,59 +23,304 @@
     <field>
         <id>action.operator</id>
-        <label>Logical operator (for ACLs)</label>
+        <label>Logical operator for conditions</label>
         <type>dropdown</type>
-        <help><![CDATA[Choose an logical operator to be used to form a condition.]]></help>
+        <help><![CDATA[Choose an logical operator.]]></help>
+    </field>
+    <field>
+        <label>HAProxy function</label>
+        <type>header</type>
     </field>
     <field>
         <id>action.type</id>
-        <label>Choose action</label>
+        <label>Run function</label>
         <type>dropdown</type>
-        <help><![CDATA[Choose an action that should be executed if the condition is true.]]></help>
+        <help><![CDATA[Choose an HAProxy function that should be executed if the condition evaluates to true.]]></help>
     </field>
     <field>
-        <label>Optional parameters</label>
+        <label>Parameters</label>
         <type>header</type>
+        <style>type_table table_use_backend</style>
     </field>
     <field>
-        <id>action.useBackend</id>
-        <label>Use backend</label>
+        <id>action.use_backend</id>
+        <label>Use backend pool</label>
         <type>dropdown</type>
-        <help><![CDATA[Use this backend if the condition is true.<br/><b>Not used for any other action.</b>]]></help>
+        <help><![CDATA[HAProxy will use this backend pool if the condition evaluates to true.]]></help>
     </field>
     <field>
-        <id>action.useServer</id>
+        <label>Parameters</label>
+        <type>header</type>
+        <style>type_table table_use_server</style>
+    </field>
+    <field>
+        <id>action.use_server</id>
         <label>Use server</label>
         <type>dropdown</type>
-        <help><![CDATA[Use this server if the condition is true.<br/><b>Not used for any other action.</b>]]></help>
+        <help><![CDATA[HAProxy will use this server if the condition evaluates to true.]]></help>
+    </field>
+    <field>
+        <label>Parameters</label>
+        <type>header</type>
+        <style>type_table table_http-request_auth</style>
+    </field>
+    <field>
+        <id>action.http-request_auth</id>
+        <label>Auth Realm</label>
+        <type>text</type>
+        <help><![CDATA[When HAProxy requests user name and password from the user, this optiomal authentication realm is returned with the response (typically the application's name).]]></help>
+    </field>
+    <field>
+        <label>Parameters</label>
+        <type>header</type>
+        <style>type_table table_http-request_redirect</style>
+    </field>
+    <field>
+        <id>action.http-request_redirect</id>
+        <label>HTTP Redirect</label>
+        <type>text</type>
+        <help><![CDATA[Use HAProxy's powerful redirect function to return a HTTP redirection. See <a href="http://cbonte.github.io/haproxy-dconv/1.7/configuration.html#redirect">HAProxy's documentation</a> for further details and examples.]]></help>
```

Comment:
Links in the help text will only make me through out all translations for it as mistakes are easily made by translators. "powerful" is debatable, from a pure documentation perspective we can live without it. :)

Reply:
The syntax for the redirect function is not self-explaining (to say the least). I plan to add a GUI for it, but until then I think the link is required. :-/